test/test-network/systemd-networkd-tests.py

```
@@ -0,0 +1,857 @@
+
+class Utilities():
+    dhcp_server_data = []
+
+    def read_link_attr(self, link, dev, attribute):
+        with open(os.path.join(os.path.join(os.path.join('/sys/class/net/', link), dev), attribute)) as f:
+            return f.readline().strip()
+
+    def link_exits(self, link):
+        return os.path.exists(os.path.join('/sys/class/net', link))
+
+    def link_remove(self, links):
+        for link in links:
+            if os.path.exists(os.path.join('/sys/class/net', link)):
+                subprocess.call(['ip', 'link', 'del', 'dev', link])
+
+    def read_ipv6_sysctl_attr(self, link, attribute):
+        with open(os.path.join(os.path.join(network_sysctl_ipv6_path, link), attribute)) as f:
+            return f.readline().strip()
+
+    def read_ipv4_sysctl_attr(self, link, attribute):
+        with open(os.path.join(os.path.join(network_sysctl_ipv4_path, link), attribute)) as f:
+            return f.readline().strip()
+
+    def copy_unit_to_networkd_unit_path(self, *units):
+        for unit in units:
+            shutil.copy(os.path.join(networkd_ci_path, unit), network_unit_file_path)
+
+    def remove_unit_from_networkd_path(self, units):
+        for unit in units:
+            if (os.path.exists(os.path.join(network_unit_file_path, unit))):
+                os.remove(os.path.join(network_unit_file_path, unit))
+
+    def start_dnsmasq(self):
+        subprocess.call(['dnsmasq', '-8', '/var/run/networkd-ci/test-dnsmasq-log-file',
+                         '--log-queries=extra',  '--log-dhcp', '--pid-file=/var/run/networkd-ci/test-test-dnsmasq.pid',
+                         '--conf-file=/dev/null', '--interface=veth-peer', '--enable-ra', '--dhcp-range=2600::10,2600::20',
+                         '--dhcp-range=192.168.5.10,192.168.5.200', '-R', '--dhcp-leasefile=/var/run/networkd-ci/lease',
+                         '--dhcp-option=26,1492', '--dhcp-option=option:router,192.168.5.1', '--dhcp-option=33,192.168.5.4,192.168.5.5'])
```

`check_call` or else it will ignore errors.